tests/ion_hash_tests.rs

```diff
@@ -123,7 +123,7 @@ fn ion_hash_tests() -> IonHashTestResult<()> {
 
 fn test_file(file_name: &str) -> IonHashTestResult<()> {
     let data = read(file_name).map_err(|source| ion_rs::IonError::IoError { source })?;
-    let elems = Element::read_all(&data)?;
+    let elems = Element::read_all(data)?;
```

Comment:
What's going on here?

Reply:
`clippy` noticed that `data` (which was already a reference) was being turned into a doubly-indirect reference unnecessarily. This change was unrelated to the core purpose of the PR.